src/components/Shared/ReceiveModal/ReceiveModal.tsx

```diff
@@ -125,46 +148,56 @@ const ReceiveModal: FC<ReceiveModalProps> = (props) => {
           </p>
         </>
       )}
+
       <form
         className="w-full flex flex-col items-center"
-        onSubmit={generateInvoiceHandler}
+        onSubmit={handleSubmit(onSubmit)}
       >
-        {isLoading && (
-          <div className="p-5">
-            <LoadingSpinner />
-          </div>
-        )}
-        {showLnInvoice && (
-          <div className="flex flex-col pb-5 justify-center text-center">
-            <AmountInput amount={amount} onChangeAmount={amountChangeHandler} />
-            <div className="flex flex-col justify-center mt-2">
-              <label
-                htmlFor="comment"
-                className="block text-gray-700 dark:text-white text-sm mb-2 font-bold text-left"
-              >
-                Comment
-              </label>
-              <input
-                id="comment"
-                type="text"
-                placeholder="Optional comment"
-                value={comment}
-                onChange={commentChangeHandler}
-                className="input-underline"
+        <fieldset className="w-4/5 mb-5">
+          {isLoading && (
+            <div className="p-5">
+              <LoadingSpinner />
+            </div>
+          )}
+
+          {showLnInvoice && (
+            <div className="flex flex-col pb-5 justify-center text-center">
+              <AmountInput
+                amount={amount}
+                register={register("amountInput", {
+                  required: t(
+                    "forms.validation.chainAmount.required"
```

Comment:
This is not included in the translation JSON or at least doesnt translate correctly)

![image](https://user-images.githubusercontent.com/9399034/143677235-40ae9135-2bfd-44d6-94ad-46487d291052.png)